test/powershell/Modules/PSDiagnostics/PSDiagnostics.Tests.ps1

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe -Name "PSDiagnostics cmdlets tests" -Tag "CI", "RequireAdminOnWindows" {
+    BeforeAll {
+        $OriginalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        if ( -not $IsWindows ) {
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+    }
+    AfterAll {
+        $Global:PSDefaultParameterValues = $OriginalDefaultParameterValues
```

Comment:
The retrieval of these values is pulled from the local scope, but you're setting it back at the global scope.

This is _probably_ fine, but I don't know if it's a good idea to risk edge cases with this mix of scoping get/set behaviour. I think it might be better to either pull both from $Global: or just both from the local scope. 🙂

Reply:
Setting  `$PSDefaultParameterValues` changes effects globally, so when setting it back we should specify the scope else a local variable will get created with the same name leaving the global variable with the change made.

other scopes also doesn't help here with as it will get effected only in `{...}`

Reply:
This is a slightly trick. (1) You could have created this variable in another scope, but why? (2) You need to restore the original value and we just use "try-finally" style that is `AfterAll` and just use it in the most reliable way - in global scope.

Reply:
And seems another scope doesn't work.

Reply:
Good to know, thank you. Seems... odd, though, to have such a mismatch. 😄